### ISSUE TITLE
Is there a predictor?

### ISSUE THREAD

**Issue by jloveric** [revalo/tree-diffusion#6]
Do you have a simple predictor somewhere? Looking for something that takes in a simple image and then predicts the program using your model.

**Comment by revalo** [comment]
Not yet, I don't expect this codebase to be actually useful for turning images into programs because the DSLs are very small and narrow. If you still want to try it you can look at `scripts/td_search_eval.py`

The codebase is meant mostly for academic use on the technique and reproducibility + transparency of evaluation numbers in the paper.

I am working on making it easier to use it on a custom image! I just don't expect it to work on arbitrary images very well, because the DSLs just can't express them.

**Comment by jloveric** [comment]
Turns out we have a problem so "simple", what you have might be sufficient.

**Comment by jloveric** [comment]
Here is an example of an image I want to fit. Should it be able to fit this one? It seems to get sort of close, but doesn't seem to be able to solve it. It's hard to see here, but it's just a rectangle inside a 128x128 image.
![target_0](https://github.com/revalo/tree-diffusion/assets/5200721/4b5cfab2-7b9b-44b2-9a9a-26a404443508)

**Comment by debuggity** [comment]
> Here is an example of an image I want to fit. Should it be able to fit this one? It seems to get sort of close, but doesn't seem to be able to solve it. It's hard to see here, but it's just a rectangle inside a 128x128 image. ![target_0](https://private-user-images.githubusercontent.com/5200721/336907853-4b5cfab2-7b9b-44b2-9a9a-26a404443508.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc2MzU2NDUsIm5iZiI6MTcxNzYzNTM0NSwicGF0aCI6Ii81MjAwNzIxLzMzNjkwNzg1My00YjVjZmFiMi03YjliLTQ0YjItOWE5YS0yNmE0MDQ0NDM1MDgucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDYwNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA2MDZUMDA1NTQ1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MTYxN2NjMzVkNDEwMTgzY2Y2OTNmMTc3YjgwYjE5NzQyMmU5ZDdjZjEyODkwZmUwMWEyYzU5NjU3ZDE4MDlkOSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.VRDjeyWGQDCxY8uiRUqjAMe_XIquxNYRPEP8P-Uw6l8)

You might be able to solve a problem like that using a simple genetic algorithm to be honest.

**Comment by jloveric** [comment]
Yeah, we already solve it a different, very simple way, but the algorithm in this paper could lead to something better for more complicated situations.

**Comment by revalo** [comment]
I'm working on making this repository actually useful for new images! Stay tuned. I sadly fell sick right after release so it's been a bit slow.

**Comment by Emasoft** [comment]
@revalo Please get well. Your paper is very promising. Definitely a step in the right direction. Making the repo code to work well would be awesome to do research into code diffusion.
